### PR TITLE
Add support for displaying time listeners

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Add support for displaying time pattern listeners
Supports https://github.com/home-assistant/core/pull/41147

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bdraco** [comment]
Closing per https://github.com/home-assistant/core/pull/41147#issuecomment-703250497

File: src/data/ws-templates.ts [hunk 1]
```diff
@@ -5,10 +5,17 @@ export interface RenderTemplateResult {
...
+  hour: string | number;
+  minute: string | number;
+  second: string | number;
+}
+
 interface TemplateListeners {
   all: boolean;
   domains: string[];
   entities: string[];
+  time_patterns: TimePattern[];
```
Comment (bdraco) [inline src/data/ws-templates.ts:18]:
```suggestion
  time: boolean;
```

File: src/data/ws-templates.ts [hunk 2]
```diff
@@ -5,10 +5,17 @@ export interface RenderTemplateResult {
   listeners: TemplateListeners;
 }
 
+interface TimePattern {
+  hour: string | number;
+  minute: string | number;
+  second: string | number;
+}
+
```
Comment (bdraco) [inline src/data/ws-templates.ts:13]:
```suggestion
```

File: src/panels/lovelace/cards/hui-markdown-card.ts [hunk 3]
```diff
@@ -144,7 +144,7 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
     } catch (_err) {
       this._templateResult = {
         result: this._config!.content,
-        listeners: { all: false, domains: [], entities: [] },
+        listeners: { all: false, domains: [], entities: [], time_patterns: [] },
```
Comment (bdraco) [inline src/panels/lovelace/cards/hui-markdown-card.ts:147]:
```suggestion
        listeners: { all: false, domains: [], entities: [], time: false },
```

File: src/translations/en.json [hunk 4]
```diff
@@ -2847,6 +2847,7 @@
             "jinja_documentation": "Jinja2 template documentation",
             "template_extensions": "Home Assistant template extensions",
             "unknown_error_template": "Unknown error rendering template",
+            "time_pattern": "This template listens for time patterns matching:",
```
Comment (bdraco) [inline src/translations/en.json:2850]:
```suggestion
            "time": "This template updates every 60 seconds after the last state changed event.",
```

**Comment by bdraco** [comment]
Screen Shot 2020-10-16 at 12 31 16 PM

**Review by bramkragten — APPROVED** [review]

**Pull request merged**